tests/unit/threads/resource_manager.cpp

```diff
@@ -0,0 +1,239 @@
+//  Copyright (c) 2016-2017 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// This is more of an integration test than a unit test. The goal is to verify
+// that the resource manager used to assign processing units to the executors
+// does the right thing.
+
+#include <hpx/config.hpp>
+#include <hpx/hpx_init.hpp>
+#include <hpx/include/parallel_executors.hpp>
+#include <hpx/runtime/threads/resource_manager.hpp>
+#include <hpx/util/lightweight_test.hpp>
+
+#include <boost/atomic.hpp>
+
+#include <cstddef>
+#include <string>
+#include <vector>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace test
+{
+    struct dummy_parameters
+      : hpx::parallel::executor_parameters_tag
+    {
+        dummy_parameters() {}
+    };
+
+    static const dummy_parameters dummy;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+void verify_resource_allocation(std::size_t num_execs, std::size_t num_pus)
+{
+    std::vector<hpx::threads::resource_allocation> alloc_data =
+        hpx::threads::get_resource_allocation();
+
+    HPX_TEST_EQ(num_execs, alloc_data.size());
+    for (hpx::threads::resource_allocation const& data : alloc_data)
+    {
+        HPX_TEST_EQ(num_pus, data.core_ids_.size());
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+template <typename Executor>
+void test_executors(std::size_t processing_units, std::size_t num_pus)
+{
+    boost::atomic<std::size_t> count_invocations(0);
+    std::size_t const num_tasks = 100;
+
+    typedef hpx::parallel::executor_traits<Executor> traits;
+    typedef hpx::parallel::executor_information_traits<Executor> infotraits;
+
+    std::size_t num_execs = processing_units / num_pus;
+
+    {
+        // create as many executors as we have cores available using num_pus
+        // processing unit each
+        std::vector<Executor> execs;
+        for (std::size_t i = 0; i != num_execs; ++i)
+            execs.push_back(Executor(num_pus));
+
+        verify_resource_allocation(num_execs, num_pus);
+
+        // give executors a chance to get started
+        hpx::this_thread::yield();
+
+        for(Executor& exec : execs)
+        {
+            HPX_TEST_EQ(infotraits::processing_units_count(exec, test::dummy),
+                num_pus);
+        }
+
+        // schedule a couple of tasks on each of the executors
+        for(Executor& exec : execs)
+        {
+            for (int i = 0; i != num_tasks; ++i)
+            {
+                traits::apply_execute(exec,
```

Comment:
This code will go away once this PR is merged to master, it's old code not touched by this PR.